packages/pwa/app/utils/test-utils.js

```diff
@@ -152,14 +166,21 @@ export const renderWithProviders = (children, options) =>
     })
 
 /**
- * This is used to obtain the URL pathname that would include
- * or not include the locale shortcode in the URL according to
- * the locale type configuration set in the pwa-kit.config.json
- * file.
+ * This is used to construct the URL pathname that would include
+ * or not include the default locale and site identifiers in the URL according to
+ * their configuration set in config/mocks/default.json file.
+ *
  * @param path The pathname that we want to use
- * @returns {`${string|string}${string}`} URL pathname for the given path
+ * @returns {string} URL pathname for the given path
  */
-export const getPathname = (path) => {
-    const {locale: localeType} = mockConfig.url
-    return `${localeType === urlPartPositions.PATH ? `/${DEFAULT_LOCALE}` : ''}${path}`
+export const getDefaultPathname = (path) => {
```

Comment:
This function name is weird. If I was to guess what this function returned based on it's name, I would say that it returned a path and it always returns that same path because it's the default. 

A better name might be something like `createPathWithDefaults` of `createPathWithContext`. 

But a bigger problem is that you are essentially testing functions by comparing their values with those created with the same function. 

For example in this test from above:

`expect(history.push).toHaveBeenCalledWith(getDefaultPathname('/account'))`

The link that gets clicked has its `to` param generated by `buildPathWithUrlConfig`, but you are testing that the path generated is the same as a path generated by `getDefaultPathname` which internally uses `buildPathWithUrlConfig`. 

Do you see what I mean?

Ideally the test looks something more like: 

`expect(history.push).toHaveBeenCalledWith('/us/en/account')`

Because we know the current config and we know what the url should look like. Furthermore, if we have tests for `buildPathWithUrlConfig` we have coverage for the most part in this test. 

Lets talk about this tomorrow.

Reply:
Thanks for the comment, I agree with your points. My intention for this function is to have a util that will be able to create the basePath that matches the urlConfig in the mockConfig since we use it as a global mock for all tests. You were right that it should be used as an expected value for an assertion. 🤔